packages/search/src/file-type.ts

```diff
@@ -106,8 +108,15 @@ export const isLargeFile = async (fileName: string): Promise<boolean> => {
   return fileSize > largeFileThreshold();
 };
 
-export const isBinaryFile = (fileName: string): boolean => {
-  return BINARY_FILE_EXTENSIONS.some((ext) => fileName.endsWith(ext));
+export const isBinaryFile = (filePath: string): boolean => {
+  if (BINARY_FILE_EXTENSIONS.some((ext) => filePath.endsWith(ext))) return true;
+  try {
```

Comment:
One suggestion here - if the file has an extension and the extension is determined by the binary detector to be binary or non-binary at some point, cache that result. 

We could even get rid of some of the hard coded extensions in the list if we wanted to. 

Files with no extension would always be tested by the binary checker.

Reply:
It's a good idea, but I'm not sure if it's worth the trouble. Sniffing seems to be rather cheap (it only reads up to 512 bytes of the file) and the hardcoded list of extensions should knock off most potential offenders before it even gets there.